src/rgw/rgw_op.cc

```diff
@@ -1041,11 +1041,6 @@ void RGWGetObjTags::execute(optional_yield y)
   s->object->set_atomic(s->obj_ctx);
 
   op_ret = s->object->get_obj_attrs(s->obj_ctx, y, this);
-  if (op_ret < 0) {
```

Comment:
This block returns to the caller immediately once an exception is thrown. It does not call send_respose_data function where we dump the error. 

Above check is neither used in put-object-tagging or delete-object-tagging

Reply:
if you get op_ret < 0 then it should terminate request and send response to client, line 1045-150 should be skipped right?

Reply:
Yes

Reply:
> this is just logging the error, why to avoid ? and anyways we are returning with or without this condition !

This was wrong. Code is executing further, line 1058-1064. In case of non-existing objects, this further code may not have any impact, "send_response_data(tags_bl)" will just execute with existing bufferlist !?

We can just put send_response_data(tags_bl) in this if condition then?

Reply:
We are already logging the same in sal layer as well.. 
Put and delete object tagging also just call the method and does not log the error in op.cc.
Followed the similar pattern

Reply:
As discussed. we can modify like

> if (op_ret == 0) {
  attrs = s->object->get_attrs();
  auto tags = attrs.find(RGW_ATTR_TAGS);
  if(tags != attrs.end()){
    has_tags = true;
    tags_bl.append(tags->second);
  }
  send_response_data(tags_bl);

Reply:
Done, modified and tested.